spring-kafka/src/main/java/org/springframework/kafka/annotation/KafkaListenerAnnotationBeanPostProcessor.java

```diff
@@ -305,6 +306,9 @@ public void afterSingletonsInstantiated() {
 
 		// Actually register all listeners
 		this.registrar.afterPropertiesSet();
+		Map<String, ContainerGroupSequencer> sequencers =
+				this.applicationContext.getBeansOfType(ContainerGroupSequencer.class, false, false);
+		sequencers.values().forEach(seq -> seq.initialize());
```

Comment:
Doesn't look like you do anything else but just that `initialize`.
Why `ContainerGroupSequencer` can't implement a `SmartInitializingSingleton` then instead?

Reply:
I tried that but there is no way to control the order, and the `ContainerGroupSequencer.aSI()` ran before this one - the `afterPropertiesSet()` creates the group beans so we found none there; so I had to move it here.